### PR TITLE
Use react-overlays on ActionMenu

### LINKED ISSUE #505: Use react-overlays on ActionMenu
https://github.com/react-bootstrap/react-overlays

Should fix https://github.com/vtex/styleguide/issues/499, and serves as a trial for this component to be used elsewhere (e.g. tooltip, modal)

cc @klzns

### PR DESCRIPTION
**Pull request by ericreis**
This PR changes the way the `Menu` component is added to the DOM. Now it uses [`react-overlay`](https://github.com/react-bootstrap/react-overlays) to add the `Menu`to the bottom of the DOM.

This closes #505.

### CODE REVIEW

File: react/components/ActionMenu/README.md [hunk 1]
```diff
@@ -130,13 +130,13 @@ const options = [
...
-}));
+}))
 
-<div className="flex">
+;<div className="flex">
   <div className="flex flex-column">
     <div className="ma3">
       <ActionMenu
-        icon={<Cog color="currentColor"/>}
+        icon={<Cog color="currentColor" />}
```
Comment (lbebber) [inline react/components/ActionMenu/README.md]: isn't this prop now deprecated? this example should follow suit

File: react/components/ActionMenu/README.md [hunk 2]
```diff
@@ -146,7 +146,7 @@ const options = [
     </div>
     <div className="ma3">
       <ActionMenu
-        icon={<Cog color="currentColor"/>}
+        icon={<Cog color="currentColor" />}
```
Comment (lbebber) [inline react/components/ActionMenu/README.md]: ditto

File: react/modules/withForwardedRef.js [hunk 3]
```diff
@@ -1,13 +1,25 @@
...
+import PropTypes from 'prop-types'
 
-function withForwardedRef(Component) {
-  const ComponentWithRef = React.forwardRef((props, ref) => (
-    <Component {...props} forwardedRef={ref} />
-  ))
+// For more info see: https://stackoverflow.com/a/51127130/10725088
+const Element = typeof Element === 'undefined' ? function() {} : Element
+
+export const refShape = PropTypes.oneOfType([
```
Comment (lbebber) [inline react/modules/withForwardedRef.js]: should this be defined here? shouldn't this be used across multiple components that are under similar circumstances?

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,16 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Changed
+
+- **ActionMenu** now uses `react-overlays` and is being added to the DOM in
```
Comment (lbebber) [inline CHANGELOG.md]: this is an implementation detail that is not supposed to be relevant for the user; if this fixes an issue, it should be under `### Fixed` and refer only to the issue at hand

File: react/components/Menu/index.js [hunk 5]
```diff
@@ -1,8 +1,12 @@
+/* eslint-disable no-unused-vars */
```
[the remark this answers is not in the document]
Comment (ericreis) [reply react/components/Menu/index.js]: this was already removed 😄

File: react/modules/withForwardedRef.js (hunk 3 above)
Comment (ericreis) [reply react/modules/withForwardedRef.js]: this is exactly why it's defined here. this way every component that need to add `ref to the `propTypes` can just import the shape to be used.

Comment (lbebber) [reply react/modules/withForwardedRef.js]: ah sorry, my bad, I thought it was inside the `ActionMenu` component

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,6 +7,18 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Fixed
+
+- **ActionMenu** unwanted behavior that made the component not visible when inside other components that use `overflow`.
```
Comment (lbebber) [inline CHANGELOG.md]:
```suggestion
- **ActionMenu** unwanted behavior, where the menu would be cropped when inside other components that use `overflow: hidden`.
```

File: react/components/ActionMenu/index.js [hunk 7]
```diff
@@ -110,9 +117,11 @@ ActionMenu.propTypes = {
   align: PropTypes.oneOf(['right', 'left']),
   /** If should close the menu after clicking an option */
   shouldCloseOnClick: PropTypes.bool,
-  /** Respecting button props contract. */
-  buttonProps: PropTypes.shape({ ...Button.propTypes }),
-  /** Button icon */
+  /** Respecting ButtonWithIcon props contract. For more info, click
+   * <a href='#/Components/Forms/ButtonWithIcon'>here</a>
```
Comment (lbebber) [inline react/components/ActionMenu/index.js]: I'm not sure if this is the proper place for `a` tags? Posting the link itself is fine

**Comment by ericreis** [comment]
Fixes #499

**Pull request merged**